client/API.md

```diff
@@ -0,0 +1,536 @@
+| 5xx                | The server was unable to process the request, refer to the error for a reason. Repeating the request may result in a success if the server issue has been resolved. |
+
+### Errors
+
+All requests can return JSON in the following format to provide more information about a failed request.
+
+```
+{
+    "error" : "error message"
+}
+```
+
+### Query Parameters vs JSON body
+
+To make using this API a consistent and easy experience we follow one simple rule for when extra information
+about a request is found in the query parameters of the URL or when they are part of the submitted JSON body.
+
+Query parameters are used only for GET requests and all POST requests expect parameters to be specified in the JSON body.
+
+## Tasks
+
+A task represents work for Kapacitor to perform.
+A task is defined by its name, type, TICKscript, and list of database retention policy pairs it is allowed to access.
+
+### Define Task
+
+To define a task POST to the `/task/TASK_NAME` endpoint.
+If no task exists it with that name it will be created.
+Otherwise the existing task will be updated with the provided information.
+
+Define a task using a JSON object with the following options:
+
+| Parameter | Purpose                                                                |
+| --------- | -------                                                                |
+| type      | The task type: `stream` or `batch`.                                    |
+| dbrps     | List of database retention policy pairs the task is allowed to access. |
+| script    | The content of the script.                                             |
+
+If any option is missing it will be left unmodified if the task already exists.
+
+#### Example
+
+```
+POST /task/TASK_NAME
+{
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream\n    |from()\n        .measurement('cpu')\n"
+}
+```
+
+Modify only the dbrps of the task.
+
+```
+POST /task/TASK_NAME
+{
+    "dbrps": [{"db": "NEW_DATABASE_NAME", "rp" : "NEW_RP_NAME"}]
+}
+```
+
+>NOTE: Setting any DBRP will overwrite all stored DBRPs.
+
+#### Response
+
+| Code | Meaning |
+| ---- | ------- |
+| 204  | Success |
+
+
+### Get Task
+
+To get information about a task make a GET request to the `/task/TASK_NAME` endpoint.
+
+| Query Parameter | Default | Purpose                                                                                                                                                            |
+| --------------- | ------- | -------                                                                                                                                                            |
+| dot-labels      | false   | Return the DOT formatted string using only label attributes. This is useful if you intend to render a diagram from the DOT content. Otherwise it is less readable. |
```

Comment:
Instead of bools I think there are other options that will help for the future:

1. ?fields=dot-labels&fields=stat  There could also be defaults
2. ?view=all or ?view=default or ?view=foo where a few is a named list of fields.